opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryMetricsExporter.java

```diff
@@ -0,0 +1,210 @@
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+
+public class OpenTelemetryMetricsExporter extends MetricExporter {
+  private static final Logger LOGGER =
+      Logger.getLogger(OpenTelemetryMetricsExporter.class.getName());
+
+  private static final String EXPORTER_NAME = "OpenTelemetryMetricExporter";
+  private static final InstrumentationLibraryInfo INSTRUMENTATION_LIBRARY_INFO =
+      InstrumentationLibraryInfo.create("io.opentelemetry.opencensusshim", null);
+
+  private final IntervalMetricReader intervalMetricReader;
+  private final io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter;
+
+  public static OpenTelemetryMetricsExporter createAndRegister(
+      io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter) {
+    return new OpenTelemetryMetricsExporter(otelExporter, Duration.create(60, 0));
+  }
+
+  public static OpenTelemetryMetricsExporter createAndRegister(
+      io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter, Duration exportInterval) {
+    return new OpenTelemetryMetricsExporter(otelExporter, exportInterval);
+  }
+
+  private OpenTelemetryMetricsExporter(
+      io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter, Duration exportInterval) {
+    this.otelExporter = otelExporter;
+    IntervalMetricReader.Options.Builder options = IntervalMetricReader.Options.builder();
+    MetricReader reader =
+        MetricReader.create(
+            MetricReader.Options.builder()
+                .setMetricProducerManager(Metrics.getExportComponent().getMetricProducerManager())
+                .setSpanName(EXPORTER_NAME)
+                .build());
+    intervalMetricReader =
+        IntervalMetricReader.create(
+            this, reader, options.setExportInterval(exportInterval).build());
+  }
+
+  @Override
+  public void export(Collection<Metric> metrics) {
+    List<MetricData> metricData = new ArrayList<>();
+    Set<MetricDescriptor.Type> unsupportedTypes = new HashSet<>();
+    for (Metric metric : metrics) {
+      for (TimeSeries timeSeries : metric.getTimeSeriesList()) {
+        LabelsBuilder labelsBuilder = Labels.builder();
+        for (int i = 0; i < metric.getMetricDescriptor().getLabelKeys().size(); i++) {
+          if (timeSeries.getLabelValues().get(i).getValue() != null) {
+            labelsBuilder.put(
+                metric.getMetricDescriptor().getLabelKeys().get(i).getKey(),
+                timeSeries.getLabelValues().get(i).getValue());
+          }
+        }
+        Labels labels = labelsBuilder.build();
+        List<MetricData.Point> points = new ArrayList<>();
+        MetricDescriptor.Type type = null;
+        for (Point point : timeSeries.getPoints()) {
+          long timestampNanos =
```

Comment:
I'd like to see the contents of the loop extracted into a method, to make this code a little easier to understand.